### ISSUE TITLE
mbt build failed with time-out

### ISSUE THREAD

**Issue by t-byer** [SAP/cloud-mta-build-tool#528]
I tried to build a repository with mbt build. At one module, the npm install ended in a time-out.
Manual execution of npm install in the same module was successful. There were no log files provided.

SOLUTION: In the .npmrc-file **of the user**, there was another registry link written down. After removing that one, the build went further

Suggestion: Provide any sort of logs in this step, to retrace the problem better

**Comment by 1084565** [comment]
Same here, time-out failed the build of a hello-world maven project.

**Comment by ShimiT** [comment]
@1084565 @t-byer - We have just released a new version (0.3.5) which increases the default timeout, in addition, we created a new task to provide a better hint about missing registry configs.

**Comment by mikolasz** [comment]
I don't know wether I can comment here (as this is a closed issue) but we are also facing the same issue.
**mbtBuild version: 1.2.16**

Just copying the first and last raws from the log:
```
15:12:58  info  mtaBuild - Using stageName 'Central Build' from env variable
15:12:58  info  mtaBuild - Project config: '.pipeline/config.yml'
15:12:58  info  mtaBuild - Project defaults: '.pipeline/piper-defaults.yml'
15:12:58  info  mtaBuild - Ignoring custom defaults from pipeline config
15:12:59  info  mtaBuild - Fetching secrets from Vault at https://vault.tools.sap/
15:12:59  warn  mtaBuild - Could not resolve param 'mtaDeploymentRepositoryPassword' from Vault
15:12:59  info  mtaBuild - Retrieving hook configuration
15:13:01  info  mtaBuild - Successfully updated <acitveProfile> details in maven settings file : '/opt/maven/apache-maven-3.6.3/conf/settings.xml'
15:13:01  info  mtaBuild - running command: npm config get registry
15:13:01  info  mtaBuild - Discovered pre-configured npm registry registry with value https://registry.npmjs.org/
15:13:01  
15:13:01  info  mtaBuild - npm registry registry was not configured, setting it to https://int.repositories.cloud.sap/artifactory/api/npm/build-releases-npm
15:13:01  info  mtaBuild - running command: npm config set registry https://int.repositories.cloud.sap/artifactory/api/npm/build-releases-npm
15:13:01  info  mtaBuild - "mta.yaml" file found in project sources
15:13:01  info  mtaBuild - No timestamp contained in "mta.yaml". File has not been modified.
15:13:01  info  mtaBuild - Executing mta build call: "mbt build --mtar auditportal.mtar --platform CF --source ./ --target /home/jenkins/agent/workspace/Concur-MBS_auditportal_main"
15:13:01  info  mtaBuild - running command: mbt build --mtar auditportal.mtar --platform CF --source ./ --target /home/jenkins/agent/workspace/Concur-MBS_auditportal_main
15:13:01  info  mtaBuild - [2022-07-18 13:13:01]  INFO Cloud MTA Build Tool version 1.2.16
15:13:01  info  mtaBuild - [2022-07-18 13:13:01]  INFO generating the "Makefile_20220718131301.mta" file...
15:13:01  info  mtaBuild - [2022-07-18 13:13:01]  INFO done
15:13:01  info  mtaBuild - [2022-07-18 13:13:01]  INFO executing the "make -f Makefile_20220718131301.mta p=CF mtar=auditportal.mtar strict=true mode= t=\"/home/jenkins/agent/workspace/Concur-MBS_auditportal_main\"" command...
15:13:01  info  mtaBuild - [2022-07-18 13:13:01]  INFO validating the MTA project
15:13:01  info  mtaBuild - [2022-07-18 13:13:01]  INFO running the "before-all" build...
15:13:01  info  mtaBuild - [2022-07-18 13:13:01]  INFO executing the "npm install --production" command...
....
15:36:35  error mtaBuild - ......................................................................................................................................................................................................................................................................................................................................................................................................................[2022-07-18 13:36:29] ERROR could not execute the "mvn clean package" command: the process was interrupted
15:36:35  error mtaBuild - [2022-07-18 13:36:29] ERROR could not build the "auditportal-srv" module: the build timed out after 20m0s
15:36:35  info  mtaBuild - make: *** [Makefile_20220718131301.mta:37: auditportal-srv] Error 1
15:36:35  info  mtaBuild - Error: could not build the MTA project: could not execute the "make -f Makefile_20220718131301.mta p=CF mtar=auditportal.mtar strict=true mode= t=\"/home/jenkins/agent/workspace/Concur-MBS_auditportal_main\"" command: exit status 2
15:36:35  info  mtaBuild - fatal error: errorDetails{"category":"build","correlationId":"https://concurmbs.jaas-gcp.cloud.sap.corp/job/Concur-MBS/job/auditportal/job/main/118/","error":"running command 'mbt' failed: cmd.Run() failed: exit status 1","library":"SAP/jenkins-library","message":"failed to execute mta build","result":"failure","stepName":"mtaBuild","time":"2022-07-18T13:36:29.664593307Z"}
15:36:35  fatal mtaBuild - failed to execute mta build - running command 'mbt' failed: cmd.Run() failed: exit status 1
15:36:35  info  mtaBuild - persisting reports to GCS is disabled, because gcsBucketId is empty
```

Previously we solved this issue in a way to raise the timeout for the build the 20m. But now it seems that is not enough.
<img width="359" alt="image" src="https://user-images.githubusercontent.com/46314078/179530795-d8c298f4-5c7a-48fa-9f28-d6ac8183f464.png">

What I can see in the pipeline log is that it is downloading the different artifacts from the repository for minutes...
Downloaded from artifactory: [https://int.repositories.cloud.sap/artifactory/build-releases/org/apache/maven/](https://int.repositories.cloud.sap/artifactory/build-releases/org/apache/maven/enforcer/enforcer/3.0.0-M3/enforcer-3.0.0-M3.pom)

locally I can build the whole project in one minute. (no artifacts download)
